src/Microsoft.AspNetCore.Blazor.Build/targets/Publish.targets

```diff
@@ -0,0 +1,59 @@
+﻿<Project>
+  <PropertyGroup>
+    <BlazorLinkOnBuild Condition="'$(Configuration)'=='Release' AND '$(BlazorLinkOnBuild)'==''">true</BlazorLinkOnBuild>
+    <BlazorPublishDistDir>$(AssemblyName)\dist\</BlazorPublishDistDir>
+
+    <!-- Disable unwanted parts of the default publish process -->
+    <CopyBuildOutputToPublishDirectory>false</CopyBuildOutputToPublishDirectory>
+    <CopyOutputSymbolsToPublishDirectory>false</CopyOutputSymbolsToPublishDirectory>
+    <PreserveCompilationContext>false</PreserveCompilationContext>
+    <RazorCompileOnPublish>false</RazorCompileOnPublish>
+    <GenerateDependencyFile>false</GenerateDependencyFile>
+    <IsWebConfigTransformDisabled>true</IsWebConfigTransformDisabled>
+  </PropertyGroup>
+
+  <Target Name="BlazorGetCopyToPublishDirectoryItems" BeforeTargets="GetCopyToPublishDirectoryItems">
+    <ItemGroup>
+      <!-- Don't want to publish the assemblies from the regular 'bin' dir. Instead we publish ones from 'dist'. -->
+      <ResolvedAssembliesToPublish Remove="@(ResolvedAssembliesToPublish)" />
+
+      <!-- Move wwwroot files to output root -->
+      <ContentWithTargetPath Update="@(ContentWithTargetPath)" Condition="$([System.String]::new(%(TargetPath)).StartsWith('wwwroot\'))">
+        <TargetPath>$(BlazorPublishDistDir)$([System.String]::new(%(TargetPath)).Substring(8))</TargetPath>
+      </ContentWithTargetPath>
+      
+      <!-- Publish all the 'dist' files -->
+      <_BlazorGCTPDIDistFiles Include="$(OutDir)dist\**" />
```

Comment:
This kind of thing can get you in trouble with incrementalism and dirty-state when files get **deleted** or **removed**. Generally, speaking `**` is for users, not for SDK authors.

If possible you'll want to recompute this list of files from the actual inputs, and not rely on a wildcard.

Reply:
Thanks for the info! I've updated it to use the itemgroup computed by our earlier build target rather than looking at the files on disk. Wildcard gone.